File: 100-Days-Coding-Challenge/1_Count_Good_Triplets/solution.py

def solution( array : list, a :int, b:int, c:int) -> int:
    arrayLength, count = len(array), 0

    for i in range(arrayLength-2):
        for j in range(i+1, arrayLength-1):
            if abs( array[i]-array[j]) > a : break
            else:
                for k in range(j+1, arrayLength):
                    count += (abs(array[j]-array[k])<=b and abs(array[i]-array[k])<=c)
    return count

array = [3,0,1,1,9,7]
print(solution(array, 7,2,3))

File: 100-Days-Coding-Challenge/1_Count_Good_Triplets/solution.go
package main

import "fmt"
import "math"

func solution( array []int, a int, b int, c int) (int){
	arrayLength, count := len(array),0

	for i:= range array {
		for j:=i+1 ; j<arrayLength; j++{
			if math.Abs(int(array[i])-int(array[j])) > a { break }
			for k:=j+1 ; k<arrayLength; k++{
				if (math.Abs(array[j]-array[k]) <= b) & (math.Abs(array[i]-array[k])<=c) {
					count++
				}
			}
		}
	}
	return count
}

func main(){
	var a int8 = 7
	var b int8 = 2
	var c int8 = 3
	array := [...]int{3,0,1,1,9,7}
	fmt.Println(solution(array,a,b,c))
}